### PR TITLE
[script] support OTBR_MDNS env variable in script/setup

### PR DESCRIPTION
**Pull request by superwhd**
This PR is a prerequisite for running border router tests using avahi in openthread repo.

### CODE REVIEW

**Review by Irving-cl — APPROVED** [review]
LGTM :+1:

File: .github/workflows/border_router.yml [hunk 1]
```diff
@@ -60,6 +60,11 @@ jobs:
             otbr_options: "-DOT_TREL=OFF -DOT_DUA=ON -DOT_MLR=ON -DOTBR_COVERAGE=ON -DOT_SRP_SERVER=ON -DOT_ECDSA=ON -DOT_SERVICE=ON -DOTBR_DUA_ROUTING=ON"
             border_routing: 0
             cert_scripts: ./tests/scripts/thread-cert/backbone/*.py
+          - name: "Border Router Avahi Advertising Proxy"
```
Comment (simonlingoogle) [inline .github/workflows/border_router.yml:63]:
```suggestion
          - name: "Border Router (Avahi)"
```

And rename the other one "Border Router (mDNSResponder)" ?

File: script/_otbr [hunk 2]
```diff
@@ -68,8 +68,8 @@ otbr_install()
         "-DOTBR_DBUS=ON"
         "-DOTBR_DNSSD_DISCOVERY_PROXY=ON"
         "-DOTBR_SRP_ADVERTISING_PROXY=ON"
-        "-DOTBR_MDNS=mDNSResponder"
         "-DOTBR_INFRA_IF_NAME=${INFRA_IF_NAME}"
+        "-DOTBR_MDNS=${OTBR_MDNS:=mDNSResponder}"
```
Comment (simonlingoogle) [inline script/_otbr:72]:
```suggestion
        "-DOTBR_MDNS=${OTBR_MDNS:-mDNSResponder}"
```

Should be `:-` ?

File: .github/workflows/border_router.yml [hunk 3]
```diff
@@ -72,6 +77,7 @@ jobs:
       INTER_OP: 0
       INTER_OP_BBR: 0
       BORDER_ROUTING: ${{ matrix.border_routing }}
+      OTBR_MDNS: ${{ matrix.otbr_mdns }}
```
Comment (simonlingoogle) [inline .github/workflows/border_router.yml:80]:
```suggestion
```
Is this necessary?
Only need to pass it to Docker?

File: script/_otbr (hunk 2 above)
Comment (superwhd) [reply script/_otbr:72]: I'm inclined to `:=`. If `OTBR_MDNS` is used again later in the script, we don't need to set a default value again.

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍 

It's great that we can pass all TCs.

File: script/_otbr (hunk 2 above)
Comment (simonlingoogle) [reply script/_otbr:72]: SGTM

File: etc/docker/Dockerfile [hunk 4]
```diff
@@ -39,11 +39,13 @@ ARG REFERENCE_DEVICE
 ARG RELEASE
 ARG REST_API
 ARG WEB_GUI
+ARG OTBR_MDNS
```
Comment (simonlingoogle) [inline etc/docker/Dockerfile:42]: We don't use `OTBR_` prefix for other Dockerfile ARGs. Should we be consistent?

```suggestion
ARG MDNS
```

**Review by wgtdkp — APPROVED** [review]
LGTM :+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**